prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -1,70 +1,65 @@
+        data = self._get_sensor_data(host=host, username=username, password=password)
+        if not data:
             return {}
-
-        raw_sensor_data = result.data.strip().split("\n")
-        sensor_data = {}
-        sensor_data_fields = ["Sensor", "Reading", "Health", "LF", "LC", "LNC", "UNC", "UC", "UF"]
-        curr_chassis_name = ""
-        curr_chassis_data: List[Dict] = []
-        for line in raw_sensor_data:
-            line = line.strip()
-            if line.startswith("Chassis ") and line.endswith(" Status"):
-                curr_chassis_name = line[9:][:-8]
-                curr_chassis_data = []
-                continue
-            curr_sensor_data = line.split("|")
-            if len(curr_sensor_data) != 9:
-                logger.warning("Ignored line in redfish sensor output: %s", line)
-                continue
-            sensor_data_item = [entry.strip() for entry in curr_sensor_data]
-            sensor_data_map = dict(zip(sensor_data_fields, sensor_data_item))
-            if any((k != v) for k, v in sensor_data_map.items()):
-                curr_chassis_data.append(sensor_data_map)
-                sensor_data[curr_chassis_name] = curr_chassis_data
-
-        return sensor_data
-
-
-class RedfishServiceStatus(Command):
-    """Command line tool for getting redfish service status."""
-
-    prefix = ""
-    command = "rf_discover.py"
-
-    def get_service_status(self) -> bool:
-        """Return whether redfish service is available or not.
-
-        Based on the output from here:
-        https://github.com/DMTF/Redfish-Tacklebox/blob/a9bcd9bdb6621619ee260b82458c5bac20dc5f6f/scripts/rf_discover.py#L25
-        """
-        result = self()
-        if result.error:
-            logger.error(result.error)
+        output = {}
+        for chassis in data:
+            name = str(chassis["ChassisName"])
+            sensors = []
+            for sensor in chassis["Readings"]:
+                reading: str = str(sensor["Reading"]) + (sensor["Units"] or "")
+
+                sensors.append(
+                    {
+                        "Sensor": sensor["Name"],
+                        "Reading": reading,
+                        "Health": sensor["Health"] or "N/A",
+                    }
+                )
+            output[name] = sensors
+        return output
+
+    def _get_sensor_data(self, host: str, username: str, password: str) -> Optional[List[Any]]:
+        """Return sensor if sensor exists else None."""
+        sensors: Optional[List[Any]] = None
+        try:
+            redfish_obj = redfish.redfish_client(
+                base_url=host, username=username, password=password
+            )
+            redfish_obj.login(auth="session")
+            sensors = redfish_utilities.get_sensors(redfish_obj)
+            return sensors
+        except InvalidCredentialsError as err:
+            logger.exception(err)
+        except SessionCreationError as err:
+            logger.exception(err)
```

Comment:
not-blocking: If we do not add any specific logic to exception why not catch both in one step. I mean:
```suggestion
        except (InvalidCredentialsError, SessionCreationError) as err:
            logger.exception(err)
```